### ISSUE TITLE
[a11y]: Search items placeholder text fails to meet minimum color contrast ratio of 4.5:1.

### ISSUE THREAD

**Issue by anamhussain94** [carbon-design-system/carbon#19553]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11.83.0

### Automated testing tool and ruleset

Success Criterion 1.4.3 Contrast (Minimum)

### Description

<img width="934" alt="Image" src="https://github.com/user-attachments/assets/99b68e89-31f7-462b-b0e3-b61cb25f13fd" />

Search items placeholder text fails to meet minimum color contrast ratio of 4.5:1.
Same issue is reproduced in Dark mode. 
Search items placeholder text should meet minimum color contrast ratio of 4.5:1 or more so that low vision users will be able see the text clearly.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-datatable-dynamic--overview

### Steps to reproduce

1.  Navigate to carbon/design [DataTable](https://react.carbondesignsystem.com/?path=/docs/components-datatable-dynamic--overview)
2. Use a tool of your choice to check the contrast ratio between the placeholder text and background color

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
Issue [#11426](https://github.com/carbon-design-system/carbon/issues/11426) also reports a low contrast problem in the placeholder text of search fields.

**Comment by paul-balchin-ibm** [comment]
Relevant W3C/WAI documentation on placeholder contrast and possible solutions.

https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text

**Comment by tay1orjones** [comment]
We'd like design to review the ideas here and certify which token/value we should use

**Comment by thyhmdo** [comment]
@tay1orjones it should be using `$text-placeholder` (Gray 40), and it looks like it needs a double check on the color value in the code 

<img width="360" height="26" alt="Image" src="https://github.com/user-attachments/assets/1042be83-0320-4e31-8fa2-b952aab0a9b5" />

-----
Related issue: https://github.com/carbon-design-system/carbon/issues/16196
From what Paul provided above: 

>Placeholder text is usually displayed with lower color contrast than text provided by users, and it disappears from form fields when users start entering text. If the placeholder text contains instructional information or examples that disappear, it makes it more difficult for users to check their responses before submitting the form.

> While placeholder text provides valuable guidance for many users, placeholder text is not a replacement for labels.

**Comment by laurenmrice** [comment]
The color token being used for the placeholder text in the Search component is `$text-placeholder`. 

From an accessibility standpoint, placeholder text isn’t required to meet color contrast standards. It’s intended to offer optional hints or examples to help users understand what to enter in the field. Because placeholder text disappears once the user begins typing, it should not replace a visible label or include essential information needed to type in a correct value in the input. This is currently working as intended and not failing a11y requirements, so we will be closing this issue.
